src/utilcode/longfilepathwrappers.cpp

```diff
@@ -0,0 +1,1317 @@
+
+
+    return ret;
+}
+
+DWORD
+GetLongPathNameWrapper(
+        _In_ LPCWSTR lpszShortPath,
+        SString& lpszLongPath
+        )
+{
+    DWORD ret = FALSE;
+    HRESULT hr = S_OK;
+    DWORD lastError;
+
+    EX_TRY
+    {
+        LongPathString shortPath(LongPathString::Literal, lpszShortPath);
+
+        if (SUCCEEDED(LongFile::NormalizePath(shortPath)))
+        {
+            COUNT_T size = lpszLongPath.GetUnicodeAllocation();
+            size         = size > 0 ? size : MAX_PATH;
+
+            ret = GetLongPathNameW(
+                    shortPath.GetUnicode(),
+                    lpszLongPath.OpenUnicodeBuffer(size - 1),
+                    (DWORD)size
+                    );
+
+            if (ret > size)
+            {
+                lpszLongPath.CloseBuffer();
+                ret = GetLongPathNameW(
+                        shortPath.GetUnicode(),
+                        lpszLongPath.OpenUnicodeBuffer(ret - 1),
+                        ret
+                        );
+
+            }
+
+            lpszLongPath.CloseBuffer(ret);
+        }
+        
+        lastError = GetLastError();
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+}
+
+BOOL
+CreateDirectoryWrapper(
+        _In_ LPCWSTR lpPathName,
+        _In_opt_ LPSECURITY_ATTRIBUTES lpSecurityAttributes
+        )
+{
+    HRESULT hr = S_OK;
+    BOOL ret   = FALSE;
+    DWORD lastError;
+
+    EX_TRY
+    {
+        LongPathString path(LongPathString::Literal, lpPathName);
+
+        if (SUCCEEDED(LongFile::NormalizePath(path)))
+        {
+            ret = CreateDirectoryW(
+                    path.GetUnicode(),
+                    lpSecurityAttributes
+                    );
+        }
+            
+        lastError = GetLastError();
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+
+}
+
+BOOL
+RemoveDirectoryWrapper(
+        _In_ LPCWSTR lpPathName
+        )
+{
+    HRESULT hr = S_OK;
+    BOOL ret   = FALSE;
+    DWORD lastError;
+    EX_TRY
+    {
+        LongPathString path(LongPathString::Literal, lpPathName);
+
+        if (SUCCEEDED(LongFile::NormalizePath(path)))
+        {
+            ret = RemoveDirectoryW(
+                    path.GetUnicode()
+                    );
+        }
+        
+        lastError = GetLastError();
+    }
+    EX_CATCH_HRESULT(hr);
+
+    if (hr != S_OK )
+    {
+        SetLastError(hr);
+    }
+    else if(ret == FALSE)
+    {
+        SetLastError(lastError);
+    }
+
+    return ret;
+
+}
+DWORD
+GetModuleFileNameWrapper(
+    _In_opt_ HMODULE hModule,
+    SString& buffer
+    )
+{
+    HRESULT hr = S_OK;
+    DWORD ret = FALSE;
+    DWORD lastError;
+    EX_TRY
+    {
+        COUNT_T size = buffer.GetUnicodeAllocation();
```

Comment:
This should be `COUNT_T size = buffer.GetUnicodeAllocation() + 1` to make full use of the buffer. GetUnicodeAllocation doesn't count the space allocated for null terminator, while GetModuleFileNameW wants the count to include the space for null terminator.